google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineArtifactDeploymentSource.java

```diff
@@ -31,19 +32,15 @@
 
   private AppEngineEnvironment environment;
 
-  /**
-   * Default constructor used instantiating plain Artifact Deployment sources.
-   */
-  public AppEngineArtifactDeploymentSource(@NotNull ArtifactPointer pointer) {
-    super(pointer);
-  }
+  private String projectName;
+  private String version;
 
   /**
    * Initialize the artifact deployment source given a target App Engine environment, and an
    * artifact pointer.
    */
   public AppEngineArtifactDeploymentSource(
-      @NotNull AppEngineEnvironment environment,
+      @Nullable AppEngineEnvironment environment,
```

Comment:
@joaoandremartins it's actually the other way around typically, but we're exploiting the @NotNull runtime instrumentation that IJ has.  I don't think IJs null dereferencing checks work if you don't use @Nullable